src/components/PackageDocs/Body.tsx

```diff
@@ -0,0 +1,80 @@
+import { LinkIcon } from "@chakra-ui/icons";
+import { Heading, As } from "@chakra-ui/react";
+import ChakraUIRenderer from "chakra-ui-markdown-renderer";
+import React from "react";
+import ReactDOMServer from "react-dom/server";
+import ReactMarkdown from "react-markdown";
+import rehypeRaw from "rehype-raw";
+import { sanitize } from "../../util/sanitize-anchor";
+
+type HeadingResolverProps = {
+  level: number;
+  children: React.ReactNode;
+};
+
+function Headings({ level, children }: HeadingResolverProps) {
+  const size: string = ["2xl", "xl", "lg", "md", "sm", "xs"][level - 1];
+  const elem = `h${level}` as As<any>;
+
+  // Use DOMParser to look for data attribute for link ID
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(
+    ReactDOMServer.renderToStaticMarkup(children as React.ReactElement),
+    "text/html"
+  );
+
+  const string = new XMLSerializer().serializeToString(doc);
+  console.log(string);
+
+  const dataElement = doc.querySelector(
+    "span[data-heading-title][data-heading-id]"
+  ) as HTMLElement;
+  const title =
+    dataElement?.dataset.headingTitle ??
+    React.Children.toArray(children)
+      .reduce((accum: string, child): string => {
+        if (typeof child === "string") {
+          return `${accum}${child}`;
+        }
+        return accum;
+      }, "")
+      .trim();
+  /* console.log(dataElement?.dataset); */
+  const id = dataElement?.dataset.headingId ?? sanitize(title);
```

Comment:
For follow up revision: can we separate these declarations for readability & omit the console.log?

Also, it looks like we'll want to optionally access `dataElement?.dataset` properties since we're assuming that the dataElement can be undefined